src/recipe.jl

```diff
@@ -214,3 +214,38 @@ function CairoMakie.draw_plot(scene::Scene, screen::CairoMakie.Screen, img::T) w
     end
 
 end
+
+"""
+    tex_annotation!(axis::Axis, lstring, x, y; mainfont = nothing, mathfont = nothing, scale_factor=1)
+
+Add TeX annotation to an existing Makie Axis. Under the hood, it does a few things:
+
+1. via `mathspec` LaTeX package, we set the `mainfont` and `mathfont`
+2. Render it using `tectonic_jll` and convert to an image matrix.
+3. call `Makie.scatter!` and using the image as `marker`, scale the image by `scale_factor` while preserving aspec ratio.
+
+    !!! note
+You can use `\textcolor` from `xcolor` inside the latex string.
+"""
+function tex_annotation!(axis::Axis, lstring, x, y; mainfont = nothing, mathfont = nothing, scale_factor=1)
+    texdoc = TeXDocument(
+        String(lstring), true;
+        requires = "\\RequirePackage{luatex85}",
+        preamble = """
+        \\usepackage{amsmath, xcolor}
+        \\usepackage{mathspec}
+        \\pagestyle{empty}
+        $(isnothing(mainfont) ? "" :
+        "\\setmainfont{$mainfont}[Scale=MatchLowercase, Ligatures=TeX]"
+       )
+        $(isnothing(mathfont) ? "" :
+        "\\setmathfont(Digits,Latin)[Scale=MatchLowercase]{$mathfont}"
+       )
```

Comment:
Whoa cool!  I didn't know you could do that.

Do `\bf`, `\it`, `\sc` etc work correctly with this?

Reply:
probably, we can probably have a comprehensive test of making 2*2*2 different math on the same canvase as a sanity check

two caveats:
- I can't find the default font's name
- I'm yet to fully understand how the exactly interaction between multiple `set*font` work

Reply:
The default font in Makie is "TeX Gyre Heros Makie", it's in the `assets` folder of Makie (I think).
I guess we should set the default math font to also be whichever math font relates to TeX Gyre Heros?  This is what I was trying to do earlier, in building up a set of known font families for which we have good math fonts.